translate/Main.hs

```diff
@@ -94,15 +94,22 @@ doWork path = do
     (_, _) <- liftIO $ runSimulation (doTranslate signs fullCircuit) (State $ const False)
     return ()
 
-doTranslate :: MonadIO m => [DynSignal] -> Concept (State DynSignal) (Transition DynSignal) -> StateT (State DynSignal) m ()
+doTranslate :: MonadIO m => [DynSignal] -> Concept (State DynSignal) (Transition DynSignal) DynSignal -> StateT (State DynSignal) m ()
 doTranslate signs circuit = do
-    let initStrs = map (\s -> (show s, False)) signs {- TODO: don't hard-code to false -}
-    let arcStrs = map (\(from, to) -> (show from, show to)) (arcs circuit)
-    liftIO $ putStr $ genSTG initStrs arcStrs
-    return ()
-
-outputs :: [(String, Bool)] -> [String]
-outputs = sort . nub . map fst
+    case validate signs circuit of
+        Valid -> do
+            let initStrs = map (\s -> (show s, False)) signs {- TODO: don't hard-code to false -}
+            let arcStrs = map (\(from, to) -> (show from, show to)) (arcs circuit)
+            let inputSigns = filter ((==Input) . interface circuit) signs
+            let outputSigns = filter ((==Output) . interface circuit) signs
+            let internalSigns = filter ((==Internal) . interface circuit) signs
+            liftIO $ putStr $ genSTG inputSigns outputSigns internalSigns initStrs arcStrs
+            return ()
+        UnusedSignal ss -> liftIO $ putStr $ "Error. The following signals have not been declared as a type: \n"
```

Comment:
I suggest to clarify the error message: `Error. The following signals are not declared as input, output or internal: \n`.